src/hotspot/share/classfile/javaClasses.inline.hpp

```diff
@@ -325,15 +325,15 @@ inline bool java_lang_Module::is_instance(oop obj) {
 inline int Backtrace::merge_bci_and_version(int bci, int version) {
   // only store u2 for version, checking for overflow.
   if (version > USHRT_MAX || version < 0) version = USHRT_MAX;
-  assert((jushort)bci == bci, "bci should be short");
-  return build_int_from_shorts(version, bci);
+  assert((u2)bci == bci, "bci should be short");
+  return build_int_from_shorts((u2)version, (u2)bci);
```

Comment:
How about instead, something like
```
return build_int_from_shorts((u2)version, checked_cast<u2>(bci));
```
I guess that loses the explicit message, but seems otherwise clearer to me.